src/rust/engine/fs/store/src/remote.rs

```diff
@@ -60,6 +66,28 @@ pub struct RemoteOptions {
   pub batch_api_size_limit: usize,
 }
 
+// TODO: this is probably better positioned somewhere else
+pub const REAPI_ADDRESS_SCHEMAS: [&str; 3] = ["grpc://", "grpcs://", "http://"];
+
+async fn choose_provider(options: RemoteOptions) -> Result<Arc<dyn ByteStoreProvider>, String> {
+  let address = options.cas_address.clone();
+  if REAPI_ADDRESS_SCHEMAS.iter().any(|s| address.starts_with(s)) {
+    Ok(Arc::new(reapi::Provider::new(options).await?))
+  } else if let Some(path) = address.strip_prefix("file://") {
+    // supporting local "file://" for a 'remote' store is a bit weird... but this is handy for
+    // testing
+    Ok(Arc::new(base_opendal::Provider::fs(
+      path,
+      "byte-store".to_owned(),
+      options,
+    )?))
+  } else {
+    Err(format!(
+      "Cannot initialise remote byte store provider with address {address}",
```

Comment:
This should probably indicate that it is an unsupported URI scheme.

Reply:
Yeah, fair enough. Theoretically an invalid URL should be caught by `global_options.py` parsing, but it's easy enough to improve this diagnostic (and, it'd be eventually good to have the Rust validation as the single source of truth).